content/guides/autodiscovery.md

```diff
@@ -4,37 +4,75 @@ kind: guide
+Docker is being [adopted rapidly](https://www.datadoghq.com/docker-adoption/). Platforms like Docker Swarm, Kubernetes, and Amazon's ECS make running Docker-ized services easier and more resilient by managing orchestration and replication across hosts. But all of that makes monitoring more difficult. How can you reliably monitor a service which is unpredictably shifting from one host to another?
 
-Datadog automatically keeps track of what is running where, thanks to its Autodiscovery feature. Autodiscovery allows you to define configuration templates that will be applied automatically to monitor your containers.
+The Datadog Agent can automatically track which services are running where, thanks to its Autodiscovery feature. Autodiscovery lets you define configuration templates for Agent checks and specify which container types each check should apply to. The Agent enables, disables, and recompiles static check configurations from the templates as containers come and go. When your NGINX container moves from 10.0.0.6 to 10.0.0.17, Autodiscovery helps the Agent update its NGINX check configuration with the new IP address so it can keep collecting NGINX metrics without any action on your part.
 
-## How it works
+<div class="alert alert-info">
+Autodiscovery was previously called Service Discovery. It is still known as Service Discovery in the Agent's code and in configuration options.
+</div>
 
-As we consider [the problem of monitoring Docker](https://www.datadoghq.com/blog/the-docker-monitoring-problem/), one strategy is to move from a host-centric model to a service-oriented model. To do this, we'll run the Datadog Agent as a containerized service, rather than using Datadog Agents installed across all of our hosts.
+# How it Works
 
-The Autodiscovery feature watches for Docker events like when a container is created, destroyed, started or stopped. When one of these happens, the Agent identifies which service is impacted, loads the configuration template for this image, and automatically sets up its checks.
+In a traditional non-container environment, Datadog Agent configuration is, like the environment in which it runs, static. The Agent reads check configurations from disk when it starts, and as long as it's running, it continuously applies every configured check. The configuration files are static, and any network-related options configured within them serve to identify specific instances of a monitored service. When an Agent check cannot connect to such a service, it's probably not because the service was re-homed somewhere else; either the service is down, or the check configuration was unable to reach the service to begin with. In any case, the Agent continues to attempt connecting to the service until an administrator steps in to troubleshoot.
```

Comment:
Thats a lot, of commas, in the, first, sentence.

for the second sentence, the way I read that is that if the config changes, the changes are applied right away without restarting the agent, but then you say the files are static. a bit confusing
 
When an Agent check cannot connect to such a service.... you aren't talking about a service and havent defined one.. 

either the service is down, or the check configuration was unable to reach the service to begin with. - the configuration cannot reach it, or the check cannot??

until an administrator steps in to troubleshoot... just the act of troubleshooting stops the attempt to connect? wont it still try. If the admin successfully solves the problem rather than just troubleshooting, i hope it will still attempt to connect.